test/itests/test-itests-catalog/src/test/java/ddf/catalog/test/TestSecurity.java

```diff
@@ -0,0 +1,294 @@
+            + "                    <!--Add any additional claims you want to grab for the service-->\n"
+            + "                    <ic:ClaimType Optional=\"true\" Uri=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role\"/>\n"
+            + "                    <ic:ClaimType Optional=\"true\" Uri=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier\"/>\n"
+            + "                    <ic:ClaimType Optional=\"true\" Uri=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress\"/>\n"
+            + "                    <ic:ClaimType Optional=\"true\" Uri=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname\"/>\n"
+            + "                    <ic:ClaimType Optional=\"true\" Uri=\"http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname\"/>\n"
+            + "                </t:Claims>\n" + "            </wst:SecondaryParameters>\n"
+            + "                ON_BEHALF_OF"
+            + "            <wst:RequestType>http://docs.oasis-open.org/ws-sx/ws-trust/200512/Issue</wst:RequestType>\n"
+            + "            <wsp:AppliesTo xmlns:wsp=\"http://schemas.xmlsoap.org/ws/2004/09/policy\">\n"
+            + "                <wsa:EndpointReference xmlns:wsa=\"http://www.w3.org/2005/08/addressing\">\n"
+            + "                    <wsa:Address>https://localhost:8993/services/QueryService</wsa:Address>\n"
+            + "                </wsa:EndpointReference>\n" + "            </wsp:AppliesTo>\n"
+            + "            <wst:Renewing/>\n" + "        </wst:RequestSecurityToken>\n" + "    </soap:Body>\n" + "</soap:Envelope>";
+
+    @Before
+    public void beforeTest() {
+        if (setupFailed) {
+            fail("Setup failed");
+        }
+
+        LOGGER.info("Before {}", testName.getMethodName());
+        if (!ranBefore) {
+            try {
+                setLogLevels();
+                waitForAllBundles();
+                waitForHttpEndpoint(SERVICE_ROOT + "/catalog/query");
+                ranBefore = true;
+            } catch (Exception e) {
+                LOGGER.error("Failed to setup test", e);
+                setupFailed = true;
+                fail("Failed to setup security tests: " + e.getMessage());
+            }
+        }
+        LOGGER.info("Starting {}", testName.getMethodName());
+    }
+
+    @Test
+    public void testAnonymousRestAccess() throws Exception {
+        String url = SERVICE_ROOT + "/catalog/query?q=*";
+
+        configureRestForAnonymous();
+
+        //test that anonymous works and check that we get an sso token
+        String cookie = when().get(url).then().log().all().assertThat().statusCode(200)
+                .assertThat().header("Set-Cookie", containsString("JSESSIONID")).extract().cookie("JSESSIONID");
+
+        //try again with the sso token
+        given().cookie("JSESSIONID", cookie).when().get(url).then().log().all().assertThat().statusCode(200);
```

Comment:
Do we need to verify that the cookie was actually used and that another session was not created? Or is that covered elsewhere?